packages/hardhat/contracts/SignatureNFT.sol

```diff
@@ -51,42 +81,60 @@ contract SignatureNFT is ERC721Tradable {
     }
 
     /**
-     * @dev mints a unique NFT from some user-selected piece of text (which we make into an image & store in Arweave)
-     * @param uri arweave url
-     * @param tokenId the tokenId to mint
-     * @param signature from `signer` to ensure no race condition on token ids and ensures this open collection is not unnecessarily vulnerable
+     * @notice      mints a unique NFT from some user-selected piece of text (which we make into an image & store in Arweave)
+     * @param start the word index the chosen highlight starts on
+     * @param end   the word index where the chosen highlight ends
      */
-    function mintSelected(string memory uri, uint256 tokenId, bytes calldata signature) external {
-        _verifySignature(tokenId, uri, signature);
-        _safeMint(creator, msg.sender, tokenId);
-        _setTokenURI(tokenId, uri);
+    function mintSelected(uint8 start, uint8 end) 
```

Comment:
we probably wanna add a require here for start < end & start >= 0 & end <= ESSAY_LENGTH